packages/syft/src/syft/service/worker/worker_pool_service.py

```diff
@@ -443,7 +443,7 @@ def scale(
         number: int,
         pool_id: Optional[UID] = None,
         pool_name: Optional[str] = None,
-    ):
+    ) -> Union[SyftError, SyftSuccess, List[ContainerSpawnStatus]]:
```

Comment:
We should change the code so that this returns `Union[SyftSuccess, SyftError]`. There is this line `return self.add_workers(...` which should be change to

```py
result = self.add_workers(...
if isinstance(result, SyftError):
    return result
```